Objects/longobject.c

```diff
@@ -4842,7 +4842,7 @@ long_lshift1(PyLongObject *a, Py_ssize_t wordshift, digit remshift)
     for (i = 0; i < wordshift; i++)
         z->ob_digit[i] = 0;
     accum = 0;
-    for (i = wordshift, j = 0; j < oldsize; i++, j++) {
+    for (j = 0; j < oldsize; i++, j++) {
```

Comment:
One more optimization could be to eliminate the `i` variable. E.g. something like
```
pointer = z->ob_digit[wordshift]
for (j = 0; j < oldsize; j++) {
       accum |= (twodigits)a->ob_digit[j] << remshift;
        pointer[j] = (digit)(accum & PyLong_MASK);
        accum >>= PyLong_SHIFT;
}
```